primer/brand
# Adding form components

## Summary
<img width="1251" alt="form components example in light and dark mode side by side " src="https://user-images.githubusercontent.com/13340707/188420027-37844999-a10d-4540-aa14-53b9b408d6aa.png">

Adds a `FormControl`, `Select`, `Checkbox` and `TextInput` component as part of https://github.com/github/primer/issues/1122

These components use a mixture of patterns from Primer React and Primer Rails equivalent components.

## List of notable changes:



- **added** four components for early testing: `FormControl`, `TextInput`, `Select`, `Checkbox`
- **added** docs and storybook playgrounds for individual components
- **added** layout example for GitHub Enterprise sign up form to both docs and storybook

## Out-of-scope in this pull request
- [ ] Smaller button sizes

## What should reviewers focus on?

[Documentation preview](https://primer-cd85ec0781-26139705.drafts.github.io/components/FormControl)

[Storybook preview](https://primer-cd85ec0781-26139705.drafts.github.io/storybook/?path=/story/components-forms-examples--git-hub-enterprise)

- Review proposed API for suitability
- Review component performance, which has been optimised for pure rendering with minimal internal state management
- Review accessibility through Storybook Axe plugin

## Steps to test:



1. Review the [docs](https://primer-cd85ec0781-26139705.drafts.github.io/components/FormControl) and [storybook playground](https://primer-cd85ec0781-26139705.drafts.github.io/storybook/?path=/story/components-forms-formcontrol--playground) to ensure the form control and inputs work as expected
1. Verify that FormControl and TextInput[ appears as expected in the Figma file](https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/Primer-Brand?node-id=738%3A28955).

## Supporting resources (related issues, external links, etc):

- [Primer React Components ](https://primer.style/react/FormControl/)
- [Primer CSS / Primer Rails ](https://primer.style/css/storybook/?path=/story/rails-forms-textinput--playground)

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

### New leading and trailing text props

<img width="326" alt="Screenshot 2022-08-22 at 07 52 39" src="https://user-images.githubusercontent.com/13340707/185857542-bd365d14-8d84-4cd0-8587-931057477959.png">


<img width="382" alt="Screenshot 2022-08-22 at 07 52 43" src="https://user-images.githubusercontent.com/13340707/185857539-e0f9aa71-6232-4bc9-bf35-7636d0174b24.png">
